scopus-import/src/main/java/no/sikt/nva/scopus/conversion/AffiliationGenerator.java

```diff
@@ -82,39 +75,47 @@ private List<CorporationWithContributors> fillCristinOrganizationData(
+        CorporationWithContributors corporation) {
+        return Optional.ofNullable(corporation.getScopusAuthors())
+                   .map(AuthorGroupTp::getAffiliation)
+                   .map(AffiliationTp::getCountry)
+                   .flatMap(this::searchCristinOrganization);
     }
 
-    private CorporationWithContributors searchForCristinOrganizationByName(CorporationWithContributors corporation) {
-        var organizationNames = extractOrganizationNames(corporation.getScopusAuthors());
-        var cristinOrganizations =
-            organizationNames.stream()
-                .map(this::searchCristinOrganization)
-                .flatMap(Optional::stream)
-                .toList();
-        return corporation.copy().withCristinCorporations(cristinOrganizations).build();
+    private Optional<CristinOrganization> searchForCristinOrganizationByName(CorporationWithContributors corporation) {
+        return extractOrganizationNames(corporation.getScopusAuthors()).stream()
+                   .map(this::searchCristinOrganization)
+                   .flatMap(Optional::stream)
+                   .findFirst();
     }
 
-    private CorporationWithContributors fetCristinOrganizationsById(CorporationWithContributors corporation) {
-        var organization = cristinConnection.fetchCristinOrganizationByCristinId(
-            corporation.getCristinOrganizationId());
-        List<CristinOrganization> validOrganizations = isValidOrganization(organization)
-                                                           ? List.of(organization)
-                                                           : List.of();
-        return corporation.copy().withCristinCorporations(validOrganizations).build();
+    private Optional<CristinOrganization> fetchCristinOrganizationsById(CorporationWithContributors corporation) {
+        return Optional.ofNullable(cristinConnection.fetchCristinOrganizationByCristinId(
+            corporation.getCristinOrganizationId()))
+                   .filter(this::isValidOrganization);
     }
 
     private boolean isValidOrganization(CristinOrganization organization) {
         return nonNull(organization) && isNotOtherNorwegianInstitution(organization);
     }
 
     private Optional<CristinOrganization> searchCristinOrganization(String organization) {
-        var searchResponse = cristinConnection.searchCristinOrganization(organization);
-        return searchResponse.isPresent() && hasSingleOrganization(searchResponse.get())
-                   ? Optional.ofNullable(getSingleOrganization(searchResponse.get()))
-                   : Optional.empty();
+        return cristinConnection.searchCristinOrganization(organization)
+                   .map(SearchOrganizationResponse::hits)
+                   .orElse(Collections.emptyList())
+                   .stream()
+                   .filter(cristinOrg -> cristinOrg.containsLabelWithValue(organization))
```

Comment:
what is the motivation for this filter after the search, maybe add a test that covers that case.

Reply:
```
    @DisplayName("Should create contributor with affiliation created by fetching scopus affiliation group country when could not fetch institution cristin identifier from pia and institution by institution name.")
    @Test
    void shouldCreateContributorAffiliationSearchingAuthorGroupCountryWhenNoResponseFromPiaAndSearchingInstitutionByName() {
        var document = ScopusGenerator.createWithNumberOfContributorsFromAuthorTp(1).getDocument();
        var affiliationCountry = getAffiliationCountry(document);
        var organization = mockSearchOrganizationByNameResponse(affiliationCountry);
        var contributor = contributorExtractorFromDocument(document).generateContributors()
                              .stream()
                              .collect(SingletonCollector.collect());

        var id = ((Organization) contributor.getAffiliations().getFirst()).getId();
        assertThat(id, is(equalTo(organization.id())));
    }
```
When looking up for contributor organization without having organization id, we use cristin-api search organization by name endpoint. This endpoint returns list of organizations, organization name is placed in labels. After we have fetched organizations, we attempt to find the one we need by filtering out organization with label/name which equals label we have from scopus xml file.